js/classes/entities/midground/enemies/ground_enemy_base.js

```diff
@@ -0,0 +1,199 @@
+        this.stance = EnemyBase.AVOID_STANCE;
+        this.enemy.state = "flee";
+    }
+
+    /**
+     * Makes the enemy pursue Chad. Also changes its stance to aggressive.
+     */
+    pursue() {
+        this.stance = EnemyBase.AGGRESSIVE_STANCE;
+        this.enemy.state = "pursue";
+    }
+
+    /**
+     * Calculates the distance between Chad and the enemy, comparing their bottom left corners
+     * in order to avoid having to account for character/enemy height.
+     * 
+     * @returns {number} the distance between the bottom left corners of Chad and the enemy
+     */
+    chadDistance() {
+        return Vector.distance(Vector.add(CHAD.getCenter(), new Vector(0, CHAD.scaledSize.y / 2)),
+            Vector.add(this.enemy.getCenter(), new Vector(0, this.enemy.scaledSize.y / 2)));
+    }
+    
+    /**
+     * Update the enemy. Updates its position, state variables, and handles block collision.
+     */
+    update() {
+        if (this.enemy.health > 0) {
+            let xVelocity = 0;
+            // TODO: remove multiplication by GAME.clockTick once PHYSICS.GRAVITY_ACC is reduced
+            this.enemy.yVelocity += PHYSICS.GRAVITY_ACC * GAME.clockTick;
+
+            if (this.chadDistance() < this.reactionDistance) {
+                // if Chad is within our reaction distance, react based on our stance.
+                if (this.stance === GroundEnemyBase.AGGRESSIVE_STANCE) {
+                    this.enemy.state = "pursue";
+                } else if (this.stance === GroundEnemyBase.AVOID_STANCE) {
+                    this.flee();
+                }
+            } else if (this.enemy.state != "roam") {
+                // if we don't have anything to react to anymore, start roaming
+                this.enemy.state = "roam";
+                this.targetX = this.enemy.pos.x;
+                this.minRoamX = this.enemy.pos.x - this.enemy.maxRoamDistance;
+            }
+
+            // if we're pursuing Chad, update the target position to Chad's position
+            // also avoid changing direction in the middle of an attack animation
+            if (this.enemy.state === "pursue" && this.enemy.action != "attacking") {
+                this.setTargetX(CHAD.getCenter().x);
+            }
+
+            if (Math.abs(this.targetX - this.enemy.getCenter().x) < this.enemy.scaledSize.x / 2) {
+                // if we've reached our target position and we're roaming, set a new target position
+                if (this.enemy.state === "roam") {
+                    this.setTargetX(this.minRoamX + Math.random() * 2 * this.enemy.maxRoamDistance);
+                }
+            } else {
+                // if we haven't reached our target, set xVelocity in the target's direction
+                xVelocity = this.enemy.getDirection() * this.enemy.speed;
+            }
+
+            // if xVelocity is not 0, set action to "move" (for animation)
+            if (xVelocity) {
```

Comment:
Seems odd that we have a comment here, what if we just name the variable `xVelocity` as `xVelNotZero` or similar? What are your thoughts? I'm noticing a lot of comments that I'm not even reading, the code often is pretty clear in my opinion.

Reply:
`xVelocity` is the velocity of the enemy along the x-axis, so the name `xVelNotZero` does not seem like a good one to me. Regarding comments, I agree that there may be somewhat of an excess but when writing this I tried to mirror the way some of you commented your own code.

Reply:
Gotcha. So xVelocity is actually a velocity. How does the conditional know what value it is?

Regrading the comments, that makes sense. As of a few weeks ago I moving away from commenting when the code is clear.

Reply:
0 is falsy in JS, so if `xVelocity` is 0 the condition will evaluate to false.